### PR TITLE
Product masters repository Created

### PR DESCRIPTION
**Pull request by SaranyaVive**

### CODE REVIEW

File: BTechConsultant.Data/Interfaces/IProductMasterRepository.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	{
+		Task<int> SaveMaterial(MaterialsVM obj);
+		Task<bool> DeleteMaterial(int materialId);
+		Task<List<MaterialsVM>> GetAllMaterials();
+		Task<MaterialsVM> GetMaterialById(int materialId);
+
+		Task<int> SaveProductColor(ProductColorsVM obj);
+		Task<bool> DeleteProductColor(int productColorId);
+		Task<List<ProductColorsVM>> GetAllProductColors();
+		Task<ProductColorsVM> GetProductColorById(int productColorsId);
```
Comment (enflowsoftindia) [inline BTechConsultant.Data/Interfaces/IProductMasterRepository.cs:20]: change "productColorsId" to "productColorId"

File: BTechConsultant.Data/Repositories/ProductMasterRepository.cs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+using System.Threading.Tasks;
+
+namespace BTechConsultant.Data.Repositories
+{
+    public class ProductMasterRepository : IProductMasterRepository
+    {
+        private BTechDataContext _context;
+        private readonly IMapper _mapper;
+
+        public int ProductColorId { get; private set; }
```
Comment (enflowsoftindia) [inline BTechConsultant.Data/Repositories/ProductMasterRepository.cs:20]: Remove this..

File: BTechConsultant.Data/Repositories/ProductMasterRepository.cs [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+                if (existing != null)
+                {
+                    existing.MaterialName = obj.MaterialName;
+                    await _context.SaveChangesAsync();
+                }
+                return existing.MaterialId;
+            }
+            else
+            {
+                Material material = _mapper.Map<Material>(obj);
```
Comment (enflowsoftindia) [inline BTechConsultant.Data/Repositories/ProductMasterRepository.cs:42]: Added mapper in the Mapping Profile

File: BTechConsultant.Data/Repositories/ProductMasterRepository.cs [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+            return true;
+        }
+
+        public async Task<List<ProductColorsVM>> GetAllProductColors()
+        {
+            var existing = await _context.ProductColors.ToListAsync();
+            return _mapper.Map<List<ProductColorsVM>>(existing);
+        }
+
+        public async Task<ProductColorsVM> GetProductColorById(int productColorsId)
```
Comment (enflowsoftindia) [inline BTechConsultant.Data/Repositories/ProductMasterRepository.cs:105]: Change "productColorsId" to "productColorId"

File: BTechConsultant.Data/Repositories/ProductMasterRepository.cs [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+        }
+
+        public async Task<List<FabricTypesVM>> GetAllFabricTypes()
+        {
+            var existing = await _context.FabricTypes.ToListAsync();
+            return _mapper.Map<List<FabricTypesVM>>(existing);
+
+        }
+
+        public async Task<FabricTypesVM> GetFabricTypeById(int FabricTypeId)
```
Comment (enflowsoftindia) [inline BTechConsultant.Data/Repositories/ProductMasterRepository.cs:190]: Change "FabricTypeId" to "fabricTypeId"

File: BTechConsultant.Service/Interfaces/IProductMasterServices.cs [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        Task<int> SaveMaterial(MaterialsVM obj);
+        Task<bool> DeleteMaterial(int materialId);
+        Task<List<MaterialsVM>> GetAllMaterials();
+        Task<MaterialsVM> GetMaterialById(int materialId);
+       // Task ImportMaterials(List<MaterialsVM> lstObj);
+
+        Task<int> SaveProductColor(ProductColorsVM obj);
+        Task<bool> DeleteProductColor(int productColorId);
+        Task<List<ProductColorsVM>> GetAllProductColors();
+        Task<ProductColorsVM> GetProductColorById(int ProductColorId);
```
Comment (enflowsoftindia) [inline BTechConsultant.Service/Interfaces/IProductMasterServices.cs:22]: "ProductColorId" to "productColorId"

File: BTechConsultant.Service/Interfaces/IProductMasterServices.cs [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+       // Task ImportProductColors(List<ProductColorsVM> lstObj);
+
+        Task<int> SaveProductPattern(ProductPatternsVM obj);
+        Task<bool> DeleteProductPattern(int productPatternId);
+        Task<List<ProductPatternsVM>> GetAllProductPatterns();
+        Task<ProductPatternsVM> GetProductPatternById(int ProductPatternId);
```
Comment (enflowsoftindia) [inline BTechConsultant.Service/Interfaces/IProductMasterServices.cs:28]: "ProductPatternId" to "productPatternId"

File: BTechConsultant.Service/Interfaces/IProductMasterServices.cs [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+       // Task ImportProductPatterns(List<ProductPatternsVM> lstObj);
+
+        Task<int> SaveFabricType(FabricTypesVM obj);
+        Task<bool> DeleteFabricType(int fabricTypeId);
+        Task<List<FabricTypesVM>> GetAllFabricTypes();
+        Task<FabricTypesVM> GetFabricTypeById(int FabricTypeId);
```
Comment (enflowsoftindia) [inline BTechConsultant.Service/Interfaces/IProductMasterServices.cs:34]: "FabricTypeId" to "fabricTypeId"

File: BTechConsultant.Service/Services/ProductMasterServices.cs [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            return await productmasterRepository.GetAllProductColors();
+        }
+
+        public async Task<List<ProductPatternsVM>> GetAllProductPatterns()
+        {
+           return await productmasterRepository.GetAllProductPatterns();
+        }
+
+        public async Task<FabricTypesVM> GetFabricTypeById(int FabricTypeId)
```
Comment (enflowsoftindia) [inline BTechConsultant.Service/Services/ProductMasterServices.cs:60]: "FabricTypeId" to "fabricTypeId"

File: BTechConsultant.Service/Services/ProductMasterServices.cs [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            return await productmasterRepository.GetFabricTypeById(FabricTypeId);
+        }
+
+        public async Task<MaterialsVM> GetMaterialById(int materialId)
+        {
+            return await productmasterRepository.GetMaterialById(materialId);
+        }
+
+        public async Task<ProductColorsVM> GetProductColorById(int ProductColorId)
```
Comment (enflowsoftindia) [inline BTechConsultant.Service/Services/ProductMasterServices.cs:70]: "ProductColorId" to "productColorId"

File: BTechConsultant.Data/Interfaces/IProductMasterRepository.cs (hunk 1 above)
Comment (SaranyaVive) [reply BTechConsultant.Data/Interfaces/IProductMasterRepository.cs:20]: Done

File: BTechConsultant.Data/Repositories/ProductMasterRepository.cs (hunk 2 above)
Comment (SaranyaVive) [reply BTechConsultant.Data/Repositories/ProductMasterRepository.cs:20]: done

File: BTechConsultant.Data/Repositories/ProductMasterRepository.cs (hunk 3 above)
Comment (SaranyaVive) [reply BTechConsultant.Data/Repositories/ProductMasterRepository.cs:42]: done

File: BTechConsultant.Data/Repositories/ProductMasterRepository.cs (hunk 5 above)
Comment (SaranyaVive) [reply BTechConsultant.Data/Repositories/ProductMasterRepository.cs:190]: done

File: BTechConsultant.Service/Interfaces/IProductMasterServices.cs (hunk 6 above)
Comment (SaranyaVive) [reply BTechConsultant.Service/Interfaces/IProductMasterServices.cs:22]: done

File: BTechConsultant.Service/Interfaces/IProductMasterServices.cs (hunk 7 above)
Comment (SaranyaVive) [reply BTechConsultant.Service/Interfaces/IProductMasterServices.cs:28]: done

File: BTechConsultant.Service/Interfaces/IProductMasterServices.cs (hunk 8 above)
Comment (SaranyaVive) [reply BTechConsultant.Service/Interfaces/IProductMasterServices.cs:34]: done

File: BTechConsultant.Service/Services/ProductMasterServices.cs (hunk 9 above)
Comment (SaranyaVive) [reply BTechConsultant.Service/Services/ProductMasterServices.cs:60]: done

File: BTechConsultant.Service/Services/ProductMasterServices.cs (hunk 10 above)
Comment (SaranyaVive) [reply BTechConsultant.Service/Services/ProductMasterServices.cs:70]: done

**Pull request merged**